compiler/modules/front/tests/tokenTests.bal

```diff
@@ -6,12 +6,23 @@ function test1() {
     test:assertEquals(tokenization("{xyz}"), ["{", [IDENTIFIER, "xyz"], "}"]);
     test:assertEquals(tokenization("record {|int xyz; |}"),
                 ["record", "{|", "int", [IDENTIFIER, "xyz"], ";", "|}"]);
-    test:assertEquals(tokenization("17|xyz &2..."),
+    test:assertEquals(tokenization("17|xyz &2 ..."),
                       [[DECIMAL_NUMBER, "17"], "|", [IDENTIFIER, "xyz"], "&", [DECIMAL_NUMBER, "2"], "..."]);
     test:assertEquals(tokenization(string`"abc"`),  [[STRING_LITERAL, "abc"]]);
     test:assertEquals(tokenization(string`"abc\r\n\"" | "" | "123"`),
                       [[STRING_LITERAL, "abc\r\n\""], "|", [STRING_LITERAL, ""], "|", [STRING_LITERAL, "123"]]);
     test:assertEquals(tokenization("ab.xy . f()"), [[IDENTIFIER, "ab"], ".", [IDENTIFIER, "xy"], ".",  [IDENTIFIER, "f"], "(", ")"]);
+    test:assertEquals(tokenization("32.factor()"), [[DECIMAL_FP_NUMBER, 32.0],[IDENTIFIER, "actor"], "(", ")"]);
+
+    test:assertEquals(tokenization("0.0 0.0f 0e01 .00"), [[DECIMAL_FP_NUMBER, "0.0"], [DECIMAL_FP_NUMBER, 0.0], [DECIMAL_FP_NUMBER, "0e01"], [DECIMAL_FP_NUMBER, ".00"]]);
+    test:assertEquals(tokenization("-4f 2.0"), ["-", [DECIMAL_FP_NUMBER, 4.0], [DECIMAL_FP_NUMBER, "2.0"]]);
+    test:assertEquals(tokenization("1.2.floor()"), [[DECIMAL_FP_NUMBER, "1.2"], ".", [IDENTIFIER, "floor"], "(", ")"]);
+    test:assertEquals(tokenization("1e-5 1e-5f"), [[DECIMAL_FP_NUMBER, "1e-5"], [DECIMAL_FP_NUMBER, 1.0E-5]]);
+    test:assertEquals(tokenization("1.e1 1.e1F"), [[DECIMAL_FP_NUMBER, "1.e1"], [DECIMAL_FP_NUMBER, 10.0]]);
+    test:assertEquals(tokenization(".5e10 .5e10f"), [[DECIMAL_FP_NUMBER, ".5e10"], [DECIMAL_FP_NUMBER, 5.0E9]]);
+    test:assertEquals(tokenization("-10.0e-2 -10.0e-2f"), ["-", [DECIMAL_FP_NUMBER, "10.0e-2"], "-", [DECIMAL_FP_NUMBER, 0.1]]);
+
+    test:assertEquals(tokenization(".e1"), [".", [IDENTIFIER, "e1"]]);
```

Comment:
Currently  `1.e-()` gives an error when torkenizing, perhaps it's more correct to torkenize it as `[[DECIMAL_FP_NUMBER, 1.0], [IDENTIFIER, "e"], "-", "(", ")"]]`. That may require significant change.

Reply:
Yes, that is the correct answer (except `[DECIMAL_FP_NUMBER, "1."]`.

We need to be able to do:

`1. e = 1.0`